merkledag/ipld.md

```diff
@@ -32,51 +32,73 @@ Objects with merkle-links form a Graph (merkle-graph), which necessarily is both
 
 ### What is a _merkle-path_?
 
-A _merkle-path_ is a unix-style path (e.g. `/a/b/c/d`) which initially dereferences through a _merkle-link_ and then follows _named merkle-links_ in the intermediate objects. Following a name means looking into the object, finding the _name_ and resolving the associated _merkle-link_.
+A merkle-path is a unix-style path (e.g. `/a/b/c/d`) which initially dereferences through a _merkle-link_ and allows access of elements of the referenced node and other nodes transitively.
+
+_Merkle-paths_ aren't suited for using them in a general purpose filesystem because it introduces many restrictions on file names. However, it can be used to work on special purpose filesystems. It can be compared to the `/proc` filesystem on unix computers or HTTP Web APIs where the allowed paths is restricted.
+
+General purpose filesystems are encouraged to design an object model on top of IPLD that would be specialized for file manipulation and have specific path algorithms to query this model.
+
+### How do _merkle-paths_ work?
+
+A _merkle-path_ is a unix-style path which initially dereferences through a _merkle-link_ and then follows _named merkle-links_ in the intermediate objects. Following a name means looking into the object, finding the _name_ and resolving the associated _merkle-link_.
 
 For example, suppose we have this _merkle-path_:
 
 ```
-/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k/a/b/c/d
+/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k/a/b/c/d/
 ```
 
 Where:
 - `ipfs` is a protocol namespace (to allow the computer to discern what to do)
 - `QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k` is a cryptographic hash.
 - `a/b/c/d` is a path _traversal_, as in unix.
-- this link traverses five objects.
 
-Suppose also that this path points to the object `{ "hello": "world" }`.
+Path traversal can either happen inside a single IPLD object, or can happen between objects throught *merkle-links*. The simple rule is that in case the traversal is possible in the same IPLD object, *merkle-links* should not be followed.
```

Comment:
Are we being _strict_ about always requiring `//` to traverse **merkle-links**? (that's what (8) called for)

--- 

If so, let's say: 

> Paths traversals are divided into two kinds:
> - **in-object traversals** traverse maps within the same object, and is denoted with `/`
> - **cross-object traversals** traverse across objects, _resolving through **merkle-links**_, and is denoted with `//` or with `/@link/`.
>
> We divide the traversals strictly, to avoid ambiguity in accessing properties within a **merkle-link** map itself. This is not transparent resolution, and thus a path reveals the objects it traverses. For example, `a/b//c/d//e` traverses across 3 objects.`
>
> We use `//` or `/@link/` for **cross-object traversals** depending on the filesystem implementation. For example, in unix filesystems, double slashes (`//`) are meaningless and often cleaned into a single slash (`/`). In such a case, the use of `/@link/` is required to traverse links.

---

If not, let's say:

> Paths traversals can be **in-object traversals**, which traverse maps within the same object, and **cross-object traversals**, which traverse across objects, _resolving through **merkle-links**_. 
>
> We use `/` to transparently traverse inside a single IPLD object or traverse across multiple. To avoid potential ambiguity, we MUST use **in-object traversals** wherever possible. A single slash (`/`) ALWAYS traverses **in-object** first, and **cross-object** otherwise .  A double slash (`//`) or `/@link/` ALWAYS traverses **cross-object**. For example, **merkle-links** can themselves carry properties and sub-maps. When `/` path traversals are ambiguous, they default to **in-object** (the local operation). In that case, we must use `//` or `/@link/` to traverse **cross-object**.